src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+            return "Failed"
+
+        return "Passed"
+
+    except Exception as e:
+        logger.warning("An expection has occured while performing the msi certificate check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.MSI_Cert_Check_Fault_Type, summary="Error occured while trying to pull MSI certificate")
+        cli_outputs.append("An expection has occured while performing the msi certificate check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Incomplete"
+
+
+def check_cluster_security_policy(corev1_api_instance, helm_client_location):
+
+    global cli_outputs
+
+    try:
+        # Intializing the kap_pod_present and cluster_connect_feature variable as False
+        kap_pod_present = False
+        cluster_connect_feature = False
+
+        # CMD command to get helm values in azure arc and converting it to json format
+        command = [helm_client_location, "get", "values", "azure-arc", "-o", "json"]
+
+        # Using subprocess to execute the helm get values command and fetching the output
+        response_helm_values_get = Popen(command, stdout=PIPE, stderr=PIPE)
+        output_helm_values_get, error_helm_get_values = response_helm_values_get.communicate()
+
+        if response_helm_values_get.returncode != 0:
+            if ('forbidden' in error_helm_get_values.decode("ascii") or 'timed out waiting for the condition' in error_helm_get_values.decode("ascii")):
+                telemetry.set_exception(exception=error_helm_get_values.decode("ascii"), fault_type=consts.Get_Helm_Values_Failed,
+                                        summary='Error while doing helm get values azure-arc')
+
+        # Converting output obtained in json format and fetching the clusterconnect-agent feature
+        helm_values_json = json.loads(output_helm_values_get)
+        cluster_connect_feature = helm_values_json["systemDefaultValues"]["clusterconnect-agent"]["enabled"]
+
+        # To retrieve all of the arc agent pods that are presesnt in the Cluster
+        arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+        # Traversing thorugh all agents and checking if the Kube aad proxy pod is present or not
+        for each_agent_pod in arc_agents_pod_list.items:
+
+            if(each_agent_pod.metadata.name.startswith("kube-aad-proxy")):
+                kap_pod_present = True
+                break
+
+        # Checking if any pod security policy is set
+        if(cluster_connect_feature is True and kap_pod_present is False):
+            print("Error: Unable to create Kube-aad-proxy deployment there might be a security policy present which is preventing the deployment of kube-aad-proxy as it doesn't have admin privileges.\n ")
```

Comment:
let's say - "security policy or security context constraint (SCC)"

Reply:
Made the changes.